### ISSUE TITLE
Masked TextInput does not reset after clicking "Create & create another"

### ISSUE THREAD

**Issue by intrepidws** [laravel-filament/filament#1525]
### Package

filament/filament

### Package Version

v2.10.3

### Laravel Version

v8.83.0

### Livewire Version

v2.10.1

### PHP Version

PHP 8.0.13

### Bug description

While this issue is similar to #1289, it's slightly different and I wanted to make sure that I reported it.

I have the following form on a relation manager:

```
public static function form(Form $form): Form
{
    return $form
        ->schema([
            Forms\Components\TextInput::make('title'),
            Forms\Components\TextInput::make('amount')
                ->mask(fn (Forms\Components\TextInput\Mask $mask) => $mask->money('$', ',', 2)),
        ]);
}
```

The first time that I click the "Create" button, both the `title` and `amount` fields are empty, as they should be. However, if I put values into both fields and click "Create & create another", only the `title` field resets to become empty again. It seems that the mask is preventing the `amount` field from resetting to be empty. One important thing to note is that, in this scenario, while the `amount` field appears to have a value (at least according to what is visible in the UI) when I submit the form the field is actually null.

I'm out of my league here, but I wonder if the solution might be as easy as updating the imask options via their `updateOptions()` method and then doing a field destroy via `mask.destroy()`? I'm stabbing in the dark here, but trying to be helpful.

### Steps to reproduce

1. Create a relation manager for a resource
2. Create a masked TextInput field for the related resource, like this:

```
public static function form(Form $form): Form
{
    return $form
        ->schema([
            Forms\Components\TextInput::make('title'),
            Forms\Components\TextInput::make('amount')
                ->mask(fn (Forms\Components\TextInput\Mask $mask) => $mask->money('$', ',', 2)),
        ]);
}
```

3. Click "attach" button, fill out fields and then click "Create & create another"
4. The `amount` field will retain the prior value while the `title` field will properly reset

**Comment by ryangjchandler** [comment]
Sounds like a bug in the entangle.

**Comment by danharrin** [comment]
Hah, you're right. I couldn't work this out.

**Comment by danharrin** [comment]
Fixed by #1533.

**Comment by intrepidws** [comment]
Thanks this fix worked great for this issue.
